x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/LimitedRole.java

```diff
@@ -86,7 +81,7 @@ public IndicesAccessControl authorize(String action, Set<String> requestedIndice
     @Override
     public Automaton allowedActionsMatcher(String index) {
         final Automaton allowedMatcher = super.allowedActionsMatcher(index);
-        final Automaton limitedByMatcher = super.allowedActionsMatcher(index);
+        final Automaton limitedByMatcher = limitedBy.allowedActionsMatcher(index);
```

Comment:
Gosh! Is this a bug!?

Reply:
Yes. At first sight testing for it was not trivial. The bug doesn't seem important though. I'll see what I can do.

Reply:
I've pushed a simple unit test 973896352f00623f0383ea15977970ff1494692b, it is trivial after all.

The bug is not consequential. It could manifest when resizing and when adding aliases, when using keys with role descriptors, where it fails to allow the operation, although it should.